libs/platform/src/lib/components/table/components/table-column-resizer/table-column-resizer.component.ts

```diff
@@ -0,0 +1,133 @@
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    Host,
+    HostBinding,
+    HostListener,
+    Inject,
+    NgZone,
+    OnDestroy, OnInit,
+    Optional,
+    ViewEncapsulation
+} from '@angular/core';
+import { fromEvent, Subject, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { RtlService } from '@fundamental-ngx/core/utils';
+
+import { TableColumnResizeService } from '../../table-column-resize.service';
+import { DOCUMENT } from '@angular/common';
+
+/** @dynamic */
+@Component({
+    selector: `fdp-table-column-resizer`,
+    template: `<div class="fdp-table__resizer-inner"></div>`,
+    styleUrls: ['./table-column-resizer.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        class: 'fdp-table-column-resizer'
+    }
+})
+export class PlatformTableColumnResizerComponent implements OnInit, OnDestroy {
+    /** @hidden */
+    @HostBinding('style.display')
+    get _resizerDisplay(): string {
+        if (!this._resizerPosition) {
+            return 'none';
+        }
+
+        return 'block';
+    }
+
+    /** @hidden */
+    @HostBinding('style.left')
+    get _resizerPositionLeft(): string {
+        if (this._rtl || !this._resizerPosition) {
+            return 'auto';
+        }
+
+        return this._resizerPosition + 'px';
+    }
+
+    /** @hidden */
+    @HostBinding('style.right')
+    get _resizerPositionRight(): string {
+        if (!this._rtl || !this._resizerPosition) {
+            return 'auto';
+        }
+
+        return this._resizerPosition + 'px';
+    }
+
+    /** @hidden */
+    @HostBinding('class.fdp-table-column-resizer--active')
+    get _isActive(): boolean {
+        return this._tableColumnResizeService?.resizeInProgress;
+    }
+
+    /** @hidden */
+    private get _resizerPosition(): number {
+        return this._tableColumnResizeService?.resizerPosition;
+    }
+
+    /** @hidden */
+    private _rtl = false;
+
+    /** @hidden */
+    private _listener: Subscription;
+
+    /** @hidden */
+    private _destroyed = new Subject<void>();
+
+    /** @hidden */
+    constructor(
+        private readonly _cd: ChangeDetectorRef,
+        private readonly _ngZone: NgZone,
+        @Inject(DOCUMENT) private readonly _document: Document | null,
+        @Optional() private readonly _rtlService: RtlService,
+        @Optional() @Host() private readonly _tableColumnResizeService: TableColumnResizeService,
+    ) { }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this._rtlService?.rtl
+            .pipe(takeUntil(this._destroyed))
+            .subscribe(rtl => this._rtl = rtl);
+
+        this._tableColumnResizeService?.markForCheck
+            .pipe(takeUntil(this._destroyed))
+            .subscribe(() => this._cd.markForCheck())
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._destroyed.next();
+        this._destroyed.complete();
```

Comment:
As I know we are having two valid approaches for unsubscribing. 
1) this._listener.add()  + this._listener?.unsubscribe();
2) With custom subject + pipe takeUntil. Which Platon has used. 
Personally, I prefer the second way. Because it is more concise and does not create additional nesting or actions when we use the add method.
Anyway, I think that we need to decide on an approach and choose one.

Reply:
missed to notice the second approach. I usually use the first way 😁 I agree, it would be nice to have a consistent approach seen everywhere.

Reply:
I also agree on agreeing on the approach :D

Here, the listener used to listen for `mouseup` on document to finish resize and we're listening on the document as the pointer may be not over the resizer (it's the same behavior as in the SAPUI5).

So if for some reason there was `mousedown` event, we've subscribed and then `mouseup` event didn't appear - we're unsubscribing first. But also we have `takeUntil` to stop listening if component gets destroyed earlier then `mouseup` appears.

Also added `take(1)`.